python/pyspark/ml/feature.py

```diff
@@ -2557,6 +2557,10 @@ def getFormula(self):
     def _create_model(self, java_model):
         return RFormulaModel(java_model)
 
+    def __str__(self):
+        formulaStr = self.getFormula() if self.isDefined(self.formula) else ""
```

Comment:
Technically the Scala side is `Option[String]` so prints `RFormula(None) ...` when the param is not set. Not a big deal, but `... else None` here would sync it up exactly.